extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/integration/QuarkusEnhancementContext.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.hibernate.orm.deployment.integration;
+
+import org.hibernate.bytecode.enhance.spi.DefaultEnhancementContext;
+import org.hibernate.bytecode.enhance.spi.UnloadedField;
+
+public final class QuarkusEnhancementContext extends DefaultEnhancementContext {
+
+    public static final QuarkusEnhancementContext INSTANCE = new QuarkusEnhancementContext();
+
+    private QuarkusEnhancementContext() {
+        //do not invoke, use INSTANCE
+    }
+
+    @Override
+    public boolean doBiDirectionalAssociationManagement(final UnloadedField field) {
+        //Don't enable automatic association management as it's often too surprising.
+        //Also, there's several cases in which its semantics are of unspecified,
+        //such as what should happen when dealing with ordered collections.
+        return false;
+    }
+
+    @Override
+    public ClassLoader getLoadingClassLoader() {
+        //This shouldn't matter, see: QuarkusClassFileLocator
```

Comment:
Why are we doing this, if it doesn't matter?

Reply:
good point. I guess it's a remnant of previous attempt, in which I had a custom one.. but it's also good to have a place to comment about it.
I'll replace it with throwing an exception, so to catch future misunderstandings.